File: 169-majority-element.cpp
//https://leetcode.com/problems/majority-element
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> m;
        for(auto n: nums){
            m[n]++;
        }
        int max_num = 0;
        int max_val;
        for(auto &it: m){
            if(it.second > max_num){
                max_num = it.second;
                max_val = it.first;
            }
        }

        return max_val;
    }
};

File: 169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = Counter(nums)
        return count.most_common(1)[0][0]


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m = {}
        for n in nums:
            m[n] = m.get(n, 0) + 1
        
        m_num, m_count = -1, -1
        for num, count in m.items():
            if(count > m_count):
                m_count = count
                m_num = num

        return m_num
        
